src/NServiceBus.Core/Transports/Learning/HeaderSerializer.cs

```diff
@@ -21,16 +20,16 @@ public static string Serialize(Dictionary<string, string> dictionary)
             }
         }
 
-        public static Dictionary<string, string> Deserialize(string value)
+        public static HeaderDictionary Deserialize(string value)
         {
             using (var ms = new MemoryStream(Encoding.UTF8.GetBytes(value)))
             {
-                return (Dictionary<string, string>)serializer.ReadObject(ms);
+                return (HeaderDictionary)serializer.ReadObject(ms);
             }
         }
 
         static DataContractJsonSerializer serializer = new DataContractJsonSerializer(
-            type: typeof(Dictionary<string, string>),
+            type: typeof(HeaderDictionary),
```

Comment:
would be interesting to see what's actually getting serialized this way. We'd probably have to handle this in some way to ensure this gets serialized similar to a regular DIctionary